web/client-api/src/main/java/io/deephaven/web/client/api/JsPartitionedTable.java

```diff
@@ -283,15 +271,37 @@ public Column[] getColumns() {
      *
      * @return Promise of a Table
      */
-    @JsMethod
+    @Deprecated
     public Promise<JsTable> getKeyTable() {
-        return keys.copy();
+        if (keys == null) {
+            keys = connection.newState((c, state, metadata) -> {
+                SelectOrUpdateRequest view = new SelectOrUpdateRequest();
+                view.setSourceId(baseTable.state().getHandle().makeTableReference());
+                view.setResultId(state.getHandle().makeTicket());
+                view.setColumnSpecsList(descriptor.getKeyColumnNamesList());
+                connection.tableServiceClient().view(view, metadata, c::apply);
+            }, "view only keys columns")
+                    .refetch(this, connection.metadata())
+                    .then(state -> Promise.resolve(new JsTable(state.getConnection(), state)));
+        }
+        assert keys != null;
+        return keys.then(JsTable::copy);
+    }
+
+    /**
+     * Fetch the underlying base table of the partitioned table.
+     *
+     * @return Promise of a Table
+     */
+    @JsMethod
+    public Promise<JsTable> getBaseTable() {
+        return baseTable.copy();
     }
 
     /** Close any subscriptions to underlying tables or key tables */
     private void closeSubscriptions() {
-        if (keys != null) {
-            keys.close();
+        if (baseTable != null) {
+            baseTable.close();
         }
```

Comment:
Need to release keys if not null.

If null but not resolved, we want to close it as soon as it resolves. If resolved, we're already holding the original and need to release it, any copies kept by callers will continue to function.

Roughly:
```suggestion
        if (baseTable != null) {
            baseTable.close();
        }
        if (keys != null) {
            keys.then(table -> {
                table.close();
                return Promise.resolve(table);
            });
        }
```